index.js

```diff
@@ -2787,64 +2787,62 @@
   }
   S.prop = def('prop', {a: [Accessible]}, [$.String, a, b], prop);
 
-  //# get :: Accessible a => TypeRep b -> String -> a -> Maybe b
+  //# get :: Accessible a => (b -> Boolean) -> String -> a -> Maybe c
   //.
-  //. Takes a [type representative](#type-representatives), a property
-  //. name, and an object and returns Just the value of the specified object
-  //. property if it is of the specified type (according to [`is`](#is));
+  //. Takes a predicate, a property name, and an object and returns Just the
+  //. value of the specified object property if it passes the given predicate;
   //. Nothing otherwise.
   //.
-  //. The `Object` type representative may be used as a catch-all since most
-  //. values have `Object.prototype` in their prototype chains.
-  //.
   //. See also [`gets`](#gets) and [`prop`](#prop).
   //.
   //. ```javascript
-  //. > S.get(Number, 'x', {x: 1, y: 2})
+  //. > S.get(S.is(Number), 'x', {x: 1, y: 2})
   //. Just(1)
   //.
-  //. > S.get(Number, 'x', {x: '1', y: '2'})
+  //. > S.get(S.is(Number), 'x', {x: '1', y: '2'})
   //. Nothing
   //.
-  //. > S.get(Number, 'x', {})
+  //. > S.get(S.is(Number), 'x', {})
   //. Nothing
   //. ```
-  function get(typeRep, key, obj) {
+  function get(pred, key, obj) {
     var x = obj[key];
-    return is(typeRep, x) ? Just(x) : Nothing;
+    return key in obj && pred(x) ? Just(x) : Nothing;
```

Comment:
I suppose it's cleaner that way. Though perhaps it would then be best to wrap the whole statement in an `if`. When [implementing Promise/A+](https://github.com/Avaq/Fluture/pull/35) I had to access the `.then` property *once and only once*, because property access in JavaScript is impure due to custom getters.